src/language/ast.js

```diff
@@ -508,6 +508,7 @@ export type DirectiveDefinitionNode = {
   +description?: StringValueNode,
   +name: NameNode,
   +arguments?: $ReadOnlyArray<InputValueDefinitionNode>,
+  +isRepeatable: boolean,
```

Comment:
I wonder if this should be:
```
+repeatable?: boolean,
```
If someone never uses repeatable directives, this adds no extra weight to the node.

I also don't think we need the `is` prefix: the fact that it's a boolean is enough information, and the question `repeatable? false` is close-to-valid English as-is (just as understandable as `is repeatable? false`).

Reply:
The `is` prefix primarily added for consistency with `isDeprecated`, as @IvanGoncharov pointed out https://github.com/facebook/graphql/pull/472/files/1ba4e196aa710a10eceaa346d02a970dfd0b2d3a#diff-35fb422eded6e8f4df638c0d159008f6

I think I would prefer it keep it mandatory. This makes it easier to work with and i feel that the performance overhead is negligible. But I can also make it optional.

WDYT?

Reply:
@OlegIlyenko In the context of AST, I'm more in favor of @mjmahone proposal since we already have `block` with similar prototype:
https://github.com/graphql/graphql-js/blob/0dff21fb3508191cc7efbaeff1d2e72135dd9809/src/language/ast.js#L320

Plus it looks like JS AST doesn't use `is` prefix, e.g. `async`:
https://github.com/estree/estree/blob/df2290b23a652e1ec354b6775459a3b42e22f108/es2017.md#function

But outside of AST I strongly believe it should be named `isRepeatable` for the consistency with `isDeprecated`.

> This makes it easier to work with and i feel that the performance overhead is negligible.

I'm with you on that point especially since JS parser always preserve boolean flags and they typically work with way bigger files than GraphQL.
Also defacto (in our parser) we always provide values for all keys so why should Flow typings mark that as optional.

That said I think it's outside of scope of this PR and should be disscussed for  AST as a whole and not for individual flags.
So for now I think it should be optional.

Reply:
I think both alternatives should be fine. I also considered keep `repeatable` name on AST nodes, but then renamed it with `is` prefix across the whole codebase for the sake of the naming consistency.

But I think it should totally fine to have `repeatable` field on AST node and then use `isRepeatable` everywhere else. @mjmahone What do you think?

Reply:
I renamed `isRepeatable` back to `repeatable` in all places except introspection: https://github.com/graphql/graphql-js/pull/1541/commits/89f5780fa4d5f55ab4028d2136ba7fa14b63bd2c